### PR TITLE
kubeadm: Remove the v1alpha1 API 

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:
In v1.10 and earlier, kubeadm only had the `v1alpha1` API.
In v1.11, kubeadm could read `v1alpha1` configs, but only write `v1alpha2` configs
In v1.12, kubeadm can read `v1alpha2` configs, but only write `v1alpha3` or `v1beta1` configs.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
ref: https://github.com/kubernetes/kubeadm/issues/911
Depends on: https://github.com/kubernetes/kubernetes/pull/65776

**Special notes for your reviewer**:

**Release note**:

```release-note
[action required] kubeadm: The `v1alpha1` config API has been removed.
Please convert your `v1alpha1` configuration files to `v1alpha2` using the
`kubeadm config migrate` command of kubeadm v1.11.x
```
@kubernetes/sig-cluster-lifecycle-pr-reviews

### CODE REVIEW

File: cmd/kubeadm/app/cmd/config.go [hunk 1]
```diff
@@ -161,7 +160,7 @@ func NewCmdConfigMigrate(out io.Writer) *cobra.Command {
 			locally in the CLI tool without ever touching anything in the cluster.
 			In this version of kubeadm, the following API versions are supported:
 			- %s
-			- %s
+			- TODO: kubeadm.k8s.io/v1beta1
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/config.go]: Fixed in https://github.com/kubernetes/kubernetes/pull/65629

**Review by fabriziopandini — APPROVED** [review]
Looks great!
/approve
@timothysc I let you give the final lgtm

**Comment by neolit123** [comment]
@luxas @fabriziopandini @timothysc 
hm, so if someone tries to build from master somewhere before 1.12 is released and tries to `config migrate` an old `v1alpha1` laying around, it would not work, because `v1alpha1` is removed? he/she would have to move the HEAD before these commits got in or download the 1.11.x release binaries that still have the functionality?

**Review by timothysc — CHANGES_REQUESTED** [review]
Please extract the generic cleanup work into a separate PR.  It looks like there are separate distinct changes that can/should be reviewed separately.

**Review by timothysc — APPROVED** [review]
/lgtm 

on last two commits but will require a rebase 

/hold

**Comment by luxas** [comment]
/hold cancel
Rebased on top of https://github.com/kubernetes/kubernetes/pull/65776

**Pull request merged**